Tab.js: Fixes on click handling

GeoSot opened the pull request:
* use prevent default only if triggered by anchor
* disable auto-initialization if trigger is disabled


Things did:
 * [x] add tests to check event prevent default , and change old ones to meet the `isDisabled` check during click handling
 * [x] omit `isDisabled` check on method `show`

GeoSot committed:
Tab.js: Fixes on click
* use prevent default only if triggered by anchor
* disable auto-initialization if trigger is disabled
* Remove `isDisabled` check on `show` mehtod, as it is being handled by the click handler

alpadev reviewed (commented):
Is it realistic that someone would initialize tabs by themself and call `show()` but expecting that nothing happens when the element is disabled? If not then `isDisabled` on L70 is redundant I'd say because ignoring disabled elements is handled by the click handler now.

**Edit:** Hm maybe it's not redundant. There would be a way with jQuery to ignore the isDisabled 🤔

**Another Edit:** Looks like our spec does exactly what I mentioned. Sorry @GeoSot 😅

alpadev committed:
Readd isDisabled check

Sorry @GeoSot.. Figured my proposal is breaking our spec and it would be still possible to trigger show via jQuery. Maybe it's better to leave it there even if it's somewhat redundant..

GeoSot committed:
Add tests | change existing tests to meet the new check only during click handling

XhmikosR committed:
Merge branch 'main' into gs-tabs-should-prevent-default-only-in-case-of-anchor
